core/src/main/java/com/datastax/oss/driver/internal/core/tracker/RequestLogger.java

```diff
@@ -0,0 +1,156 @@
+    boolean showValues = configProfile.getBoolean(DefaultDriverOption.REQUEST_LOGGER_VALUES);
+    int maxValues =
+        showValues ? configProfile.getInt(DefaultDriverOption.REQUEST_LOGGER_MAX_VALUES) : 0;
+    int maxValueLength =
+        showValues ? configProfile.getInt(DefaultDriverOption.REQUEST_LOGGER_MAX_VALUE_LENGTH) : 0;
+
+    logSuccess(
+        request, latencyNanos, isSlow, node, maxQueryLength, showValues, maxValues, maxValueLength);
+  }
+
+  @Override
+  public void onError(
+      Request request,
+      Throwable error,
+      long latencyNanos,
+      DriverConfigProfile configProfile,
+      Node node) {
+
+    if (!configProfile.getBoolean(DefaultDriverOption.REQUEST_LOGGER_ERROR_ENABLED)) {
+      return;
+    }
+
+    int maxQueryLength = configProfile.getInt(DefaultDriverOption.REQUEST_LOGGER_MAX_QUERY_LENGTH);
+    boolean showValues = configProfile.getBoolean(DefaultDriverOption.REQUEST_LOGGER_VALUES);
+    int maxValues =
+        showValues ? configProfile.getInt(DefaultDriverOption.REQUEST_LOGGER_MAX_VALUES) : 0;
+    int maxValueLength =
+        showValues ? configProfile.getInt(DefaultDriverOption.REQUEST_LOGGER_MAX_VALUE_LENGTH) : 0;
+    boolean showStackTraces =
+        configProfile.getBoolean(DefaultDriverOption.REQUEST_LOGGER_STACK_TRACES);
+
+    logError(
+        request,
+        error,
+        latencyNanos,
+        node,
+        maxQueryLength,
+        showValues,
+        maxValues,
+        maxValueLength,
+        showStackTraces);
+  }
+
+  @Override
+  public void close() throws Exception {
+    // nothing to do
+  }
+
+  protected void logSuccess(
+      Request request,
+      long latencyNanos,
+      boolean isSlow,
+      Node node,
+      int maxQueryLength,
+      boolean showValues,
+      int maxValues,
+      int maxValueLength) {
+
+    StringBuilder builder = formatter.logBuilder(logPrefix, node);
+    if (isSlow) {
+      formatter.appendSlowDescription(builder);
+    } else {
+      formatter.appendSuccessDescription(builder);
+    }
+    formatter.appendLatency(latencyNanos, builder);
+    formatter.appendRequest(
+        request, maxQueryLength, showValues, maxValues, maxValueLength, builder);
+    LOG.info(builder.toString());
+  }
+
+  protected void logError(
+      Request request,
+      Throwable error,
+      long latencyNanos,
+      Node node,
+      int maxQueryLength,
+      boolean showValues,
+      int maxValues,
+      int maxValueLength,
+      boolean showStackTraces) {
+
+    StringBuilder builder = formatter.logBuilder(logPrefix, node);
+    formatter.appendErrorDescription(builder);
+    formatter.appendLatency(latencyNanos, builder);
+    formatter.appendRequest(
+        request, maxQueryLength, showValues, maxValues, maxValueLength, builder);
+    if (showStackTraces) {
+      LOG.error(builder.toString(), error);
+    } else {
+      LOG.error("{} {}", builder.toString(), error.toString());
```

Comment:
Nit: would be nice to have some kind of obvious indicator/separator demarcating the builder message and the error string since it looks kinda awkward currently:

```
Error (2 ms) [0 values] SELECT release_version FROM system.local  com.datastax.oss.driver.api.core.servererrors.ServerError: test
```

Reply:
I'll also fix the double-space issue.